core/include/gnuradio-4.0/Graph.hpp

```diff
@@ -864,6 +928,111 @@ struct Graph {
                        name);
     }
 
+    void
+    processMessages(MsgPortInNamed<"__Builtin"> &port, std::span<const Message> input) {
+        // fmt::print("==>>> Calling Graph's {} processMessages and passing to children, count = {}\n", this->unique_name, input.size());
+
+        auto splitFirstComponent = [](std::string_view name) -> std::optional<std::pair<std::string_view, std::string_view>> {
+            auto itSlash = std::ranges::find(name, '/');
+            if (itSlash == name.end()) return {};
+
+            return { { std::string_view(name.begin(), itSlash), std::string_view(itSlash + 1, name.end()) } };
+        };
+
+        auto notifyChild = [this](std::string_view childName, const auto &message) {
```

Comment:
Wouldn't it be easier/better to notify every child with the same message and let the child decide whether the message is for the specific child? This way, we'd also have broadcasting-type messages.

Reply:
This implementation does support multicast.
- `target=="*"` or `target=""` - sends a message to every block
- `target=="graph_unique_name/*"` - sends a message to every block in a graph
- `target=="graph_unique_name/block_unique_name"` or `target=="*/block_unique_name"` sends to a specific block (since the names are unique globally, not only per graph)
- (not supported now, but could be easily if we need it) `target=="**/..."` for multilevel Graph skipping

I'm not a fan of spam, so sending a message to everyone and then having most of them ignore it is not a good design IMO. This provides all the features with better control and no overhead of calling `O(n)` functions that will parse a string and then bail out when only one needs to be called (with n being the number of blocks)